framework/src/main/java/org/checkerframework/framework/ajava/InsertAjavaAnnotations.java

```diff
@@ -0,0 +1,580 @@
+        if (compiler == null) {
+            System.err.println("Could not get compiler instance");
+            System.exit(0);
+        }
+
+        DiagnosticCollector<JavaFileObject> diagnostics = new DiagnosticCollector<JavaFileObject>();
+        JavaFileManager fileManager = compiler.getStandardFileManager(diagnostics, null, null);
+        if (fileManager == null) {
+            System.err.println("Could not get file manager");
+        }
+
+        CompilationTask cTask =
+                compiler.getTask(
+                        null,
+                        fileManager,
+                        diagnostics,
+                        Collections.emptyList(),
+                        null,
+                        Collections.emptyList());
+        if (!(cTask instanceof JavacTask)) {
+            System.err.println("Could not get a valid JavacTask: " + cTask.getClass());
+        }
+
+        return ((JavacTask) cTask).getElements();
+    }
+
+    /**
+     * Constructs an {@code InsertAjavaAnnotations} using the given {@code Elements} instance.
+     *
+     * @param elements an instance of {@code Elements}
+     */
+    public InsertAjavaAnnotations(Elements elements) {
+        this.elements = elements;
+    }
+
+    /** Represents some text to be inserted at a file and its location. */
+    private static class Insertion {
+        /** Offset of the insertion in characters. */
+        public int position;
+        /** The contents of the insertion. */
+        public String contents;
+        /** Whether the insertion represents an object on its own line. */
+        public boolean ownLine;
+
+        /**
+         * Constructs an insertion with the given position and contents.
+         *
+         * @param position offset of the insertion in the file
+         * @param contents contents of the insertion
+         */
+        public Insertion(int position, String contents) {
+            this(position, contents, false);
+        }
+
+        /**
+         * Constructs an insertion with the given position and contents.
+         *
+         * @param position offset of the insertion in the file
+         * @param contents contents of the insertion
+         * @param ownLine true if this insertion represents an object on its own line (doesn't
+         *     affect the contents of the insertion)
+         */
+        public Insertion(int position, String contents, boolean ownLine) {
+            this.position = position;
+            this.contents = contents;
+            this.ownLine = ownLine;
+        }
+
+        @Override
+        public String toString() {
+            return "Insertion [contents=" + contents + ", position=" + position + "]";
+        }
+    }
+
+    /**
+     * Given two JavaParser ASTs representing the same Java file but with differing annotations,
+     * stores a list of {@link Insertion}s for all annotations in the first AST into the second AST.
```

Comment:
It would be helpful to say that after a client calls some method (which one?), the client should read the `insertions` field.  Is that what this does?  Or does this just transfer annotations from one AST into the second, in which case the `insertions` field, and even the mention of `Insertion`, is an implementation detail that does not belong in the class Javadoc.